controllers/rollingupgrade_controller.go

```diff
@@ -55,6 +55,8 @@ type RollingUpgradeReconciler struct {
 	ReconcileMap        *sync.Map
 	DrainTimeout        int
 	IgnoreDrainFailures bool
+	ReplacementNodesMap sync.Map
```

Comment:
Should this be a pointer?
Also, should use just pass the pointer to RollingUpgradeContext instead of entire reconciler?

Reply:
Yes. That makes sense.

Reply:
@eytan-avisror 
Looks like it isn't allowed.
Passing on `sync.Map` as a pointer.

```
go vet ./...
# github.com/keikoproj/upgrade-manager/controllers
controllers/rollingupgrade_controller.go:177:24: literal copies lock value from r.ReplacementNodesMap: sync.Map contains sync.Mutex
controllers/helpers_test.go:77:24: literal copies lock value from r.ReplacementNodesMap: sync.Map contains sync.Mutex
make: *** [vet] Error 2
```

Reply:
Can you just do a similar implementation to that of the DrainGroup?